pkg/cgroups/cgroups.go

```diff
@@ -763,3 +764,84 @@ func HostCgroupRoot() (string, error) {
 	)
 	return "", fmt.Errorf("failed to set cgroup root: %w", err)
 }
+
+// CgroupIDFromPID returns the cgroup id for a given pid.
+func CgroupIDFromPID(pid uint32) (uint64, error) {
+	cgroupFile := fmt.Sprintf("%s/%d/cgroup", option.Config.ProcFS, pid)
+	data, err := os.ReadFile(cgroupFile)
+	if err != nil {
+		return 0, err
+	}
+
+	pathPrefix := fmt.Sprintf("%s/1/root/sys/fs/cgroup", option.Config.ProcFS)
+
+	var path string
+	v2Prefix := "0::"
+	v1Prefix := fmt.Sprintf("%d:%s:", GetCgrpSubsystemIdx(), GetCgrpControllerName())
+
+	lines := strings.Split(string(data), "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, v2Prefix) {
+			path = fmt.Sprintf("%s/%s", pathPrefix, line[len(v2Prefix):])
+			break
+		}
+		// TODO: test cgroup v1 implementation
+		if strings.HasPrefix(line, v1Prefix) {
+			path = fmt.Sprintf("%s/%s/%s", pathPrefix, GetCgrpControllerName(), line[len(v2Prefix):])
+			break
+		}
+	}
+
+	if len(path) == 0 {
+		return 0, errors.New("failed to find proper cgroup")
+	}
+
+	cgID, err := GetCgroupIdFromPath(path)
+	if err != nil {
+		return 0, err
+	}
+
+	return cgID, nil
+}
+
+// GetCgroupIdFromCgroupPath deals with some idiosyncrancies of container runtimes
+//
+// Typically, the container processes run in the cgroup path specified in the OCI spec under
+// cgroupsPath. crun, however, is an exception because it uses another directory (called subgroup)
+// under the cgroupsPath:
+// https://github.com/containers/crun/blob/main/crun.1.md#runocisystemdsubgroupsubgroup.
+//
+// This function deals with this by checking for a child directory. If it finds one (and only one)
+// it uses the cgroup id from the child.
```

Comment:
So we did some hacks https://github.com/cilium/tetragon/blob/main/pkg/sensors/exec/procevents/proc.go#L96 , but not really convincing, the old branch that uses cgroup layer to tag subdirs with same cgroupID was also result of these runtime inconsistencies... 

So If I get you correctly here, you are using cgroupID of "container" subdir right?

I would rename this function to `GetCgroupIdFromSubGroup()`   less confusion with the other one

Reply:
Done!